password_security/models/res_users.py

```diff
@@ -76,19 +55,36 @@ def _check_password_rules(self, password):
         if not password:
             return True
         company_id = self.company_id
-        password_regex = ['^']
-        if company_id.password_lower:
-            password_regex.append('(?=.*?[a-z])')
-        if company_id.password_upper:
-            password_regex.append('(?=.*?[A-Z])')
-        if company_id.password_numeric:
-            password_regex.append(r'(?=.*?\d)')
-        if company_id.password_special:
-            password_regex.append(r'(?=.*?\W)')
-        password_regex.append('.{%d,}$' % company_id.password_length)
-        if not re.search(''.join(password_regex), password):
-            raise PassError(self.password_match_message())
-        return True
+        message = []
+        if len(list(filter(str.islower, password))
+               ) < company_id.password_lower:
+            message.append('\n ' + _('Lowercase letter (At least ' +
+                                     str(company_id.password_lower) +
+                                     ' character)')
+                           )
+        if len(list(filter(str.isupper, password))
+               ) < company_id.password_upper:
+            message.append('\n ' + 'Uppercase letter (At least ' +
+                           str(company_id.password_upper) + ' character)')
+
+        if len(list(filter(str.isdigit, password))) < \
+                company_id.password_numeric:
+            message.append('\n ' + 'Numeric digit (At least ' +
+                           str(company_id.password_numeric) + ' numeric)')
+
+        if len(re.findall(r"\W", password)) < company_id.password_special:
+            message.append('\n ' + 'Special character (At least ' +
+                           str(company_id.password_special) +
+                           ' special character')
+
+        if len(password) < company_id.password_length:
+            message = ['Password must be %d characters or more.' %
```

Comment:
Why don't you put this check at the top and just use the `.append()` system as everywhere. Just to keep it consistent.

Besides, you're missing the `_()` part here.